src/app/components/nodes-view/nodes-view.component.ts

```diff
@@ -188,6 +188,20 @@ export class NodesViewComponent implements OnInit {
     return this.allocDataSource.data && this.allocDataSource.data.length === 0;
   }
 
+  formatColumn(){
+    if(this.nodeDataSource.data.length==0){
+      return
+    }
+    this.nodeColumnIds.forEach((colId)=>{
+      let emptyCell=this.nodeDataSource.data.filter((node)=>{
+        return (node as any)[colId]=="" || (node as any)[colId]=="n/a"
+      })
+      if (emptyCell.length==this.nodeDataSource.data.length){
+        this.nodeColumnIds = this.nodeColumnIds.filter(el => el!==colId)
+      }
+    })
```

Comment:
In JavaScript, Please use the[ strict equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) `===` to compare the two values.

Maybe could change to this
```ts
  formatColumn() {
    if (this.nodeDataSource.data.length === 0) {
      return;
    }
    this.nodeColumnIds.forEach((colId) => {
      const emptyCell = this.nodeDataSource.data.filter((node) => {
        return node.colId === '' || node.colId === 'n/a';
      })
      if (emptyCell.length === this.nodeDataSource.data.length) {
        this.nodeColumnIds = this.nodeColumnIds.filter(el => el !== colId);
      }
    })
  }
```

And in here https://github.com/apache/yunikorn-web/blob/master/src/app/models/node-info.model.ts#L33, add the `colId` type

```ts
export class NodeInfo {
  isSelected = false;
  constructor(
    public nodeId: string,
    public hostName: string,
    public rackName: string,
    public partitionName: string,
    public capacity: string,
    public allocated: string,
    public occupied: string,
    public available: string,
    public utilized: string,
    public colId: string,
    public allocations: AllocationInfo[] | null
  ) {}

  setAllocations(allocs: AllocationInfo[]) {
    this.allocations = allocs;
  }
}
```

Reply:
Because I saw that `node ` is defined in `NodeInfo`, so it is necessary to add the definition in `node-info.model.ts`, that we can prevent to use the `any`.

![image](https://github.com/apache/yunikorn-web/assets/22230889/668fb7e2-5508-4928-a88f-0f58ca8d4eca)

Reply:
Thank you for your clarification! I understand your concern about safety issues.

In the provided code, `colId` is a string variable that stores column headers such as `'nodeId'`, `'rackName'`, `'hostName'`, etc. This occurs during the iteration of `this.nodeColumnIds.forEach((colId)=>{})`. The objective is to retrieve attributes from the node, such as `node['nodeId']`.

However, attempting to directly use `node['nodeId']` in TypeScript triggers an error:
`No index signature with a parameter of type 'string' was found on type 'NodeInfo'.ts(7053).`
Therefore, I use `(node as any)[colId]`.

To address safety issues, we can handle it by adding a check to determine whether colId is included in NodeInfo.:
```javascript
let emptyCell=this.nodeDataSource.data.filter((node)=>{
  if (!(colId in node)) {
    console.error(`Property '${colId}' does not exist on NodeInfo.`);
    return false;
  }
  return (node as any)[colId]==="" || (node as any)[colId]==="n/a"
})
```
If there's a better alternative, please feel free to share it!

Reply:
Thanks for your suggestion. 
However, I believe there is a bug in the original code that prevents checking if all cells in the column are empty. 
Consequently, I have refactored the implementation and utilized a foreach loop to address the safety concerns you raised. 
The changes have been updated in a new commit.